planning/autoware_costmap_generator/nodes/autoware_costmap_generator/object_map_utils.cpp

```diff
@@ -32,74 +32,43 @@
-  const std::vector<std::vector<geometry_msgs::msg::Point>> & in_points,
-  const std::string & in_grid_layer_name, const int in_layer_background_value,
-  const int in_fill_color, const int in_layer_min_value, const int in_layer_max_value,
-  const std::string & in_tf_target_frame, const std::string & in_tf_source_frame,
-  const tf2_ros::Buffer & in_tf_buffer)
+void fill_polygon_areas(
+  grid_map::GridMap & out_grid_map, const std::vector<geometry_msgs::msg::Polygon> & in_polygons,
+  const std::string & in_grid_layer_name, const float in_layer_background_value,
+  const float in_fill_value, const std::string & in_tf_target_frame,
+  const std::string & in_tf_source_frame, const tf2_ros::Buffer & in_tf_buffer)
 {
   if (!out_grid_map.exists(in_grid_layer_name)) {
     out_grid_map.add(in_grid_layer_name);
   }
   out_grid_map[in_grid_layer_name].setConstant(in_layer_background_value);
 
-  cv::Mat original_image;
-  grid_map::GridMapCvConverter::toImage<unsigned char, 1>(
-    out_grid_map, in_grid_layer_name, CV_8UC1, in_layer_min_value, in_layer_max_value,
-    original_image);
-
-  cv::Mat merged_filled_image = original_image.clone();
-
-  geometry_msgs::msg::TransformStamped transform;
-  transform =
+  const geometry_msgs::msg::TransformStamped transform =
     in_tf_buffer.lookupTransform(in_tf_target_frame, in_tf_source_frame, tf2::TimePointZero);
 
-  // calculate out_grid_map position
-  grid_map::Position map_pos = out_grid_map.getPosition();
-  const double origin_x_offset = out_grid_map.getLength().x() / 2.0 - map_pos.x();
-  const double origin_y_offset = out_grid_map.getLength().y() / 2.0 - map_pos.y();
-
-  for (const auto & points : in_points) {
-    std::vector<cv::Point> cv_polygon;
-
-    for (const auto & p : points) {
-      // transform to GridMap coordinate
-      geometry_msgs::msg::Point transformed_point;
-      geometry_msgs::msg::PointStamped output_stamped, input_stamped;
-      input_stamped.point = p;
-      tf2::doTransform(input_stamped, output_stamped, transform);
-      transformed_point = output_stamped.point;
-
-      // coordinate conversion for cv image
-      const double cv_x = (out_grid_map.getLength().y() - origin_y_offset - transformed_point.y) /
-                          out_grid_map.getResolution();
-      const double cv_y = (out_grid_map.getLength().x() - origin_x_offset - transformed_point.x) /
-                          out_grid_map.getResolution();
-      cv_polygon.emplace_back(cv_x, cv_y);
+  for (const auto & poly : in_polygons) {
+    // transform from Map to GridMap coordinates
+    geometry_msgs::msg::Polygon transformed_poly;
+    tf2::doTransform(poly, transformed_poly, transform);
+    grid_map::Polygon grid_map_poly;
+    for (const auto & p : transformed_poly.points) {
+      grid_map_poly.addVertex({p.x, p.y});
+    }
+    for (grid_map_utils::PolygonIterator it(out_grid_map, grid_map_poly); !it.isPastEnd(); ++it) {
+      out_grid_map.at(in_grid_layer_name, *it) = in_fill_value;
```

Comment:
question: 
this iterator returns the index which intersects the polygon whi ch is created addVertex?

Reply:
Yes like this: https://github.com/ANYbotics/grid_map/blob/master/grid_map_core/doc/iterators/polygon_iterator.gif

Reply:
thanks! and you made faster interator! sugoi!

https://github.com/maxime-clem/autoware.universe/blob/35a4d7822882742f0ce43362d78b39f51d2b3634/common/autoware_grid_map_utils/README.md#L16-L17